homeassistant/components/apple_tv/remote.py

```diff
@@ -43,12 +49,17 @@ async def async_turn_off(self, **kwargs):
 
     async def async_send_command(self, command, **kwargs):
         """Send a command to one device."""
+        num_repeats = kwargs[ATTR_NUM_REPEATS]
+        delay = kwargs.get(ATTR_DELAY_SECS, DEFAULT_DELAY_SECS)
+
         if not self.is_on:
             _LOGGER.error("Unable to send commands, not connected to %s", self._name)
             return
 
-        for single_command in command:
-            if not hasattr(self.atv.remote_control, single_command):
-                continue
-
-            await getattr(self.atv.remote_control, single_command)()
+        for _ in range(num_repeats):
+            for single_command in command:
+                attr_value = getattr(self.atv.remote_control, single_command, None)
+                if attr_value:
```

Comment:
Should we raise an error, log an error and continue or just continue silently, if the command is missing?

Reply:
I'm happy to log an error, but I do not think we need to raise an error. The supported commands are documented, so it is more likely someone has mis-typed. None of the commands can cause a major issue with the Apple TV, so an error should suffice to enable the user to figure out what has gone wrong.

Reply:
Yeah, I think it's mostly a question of what we think is the best and most expected behavior in case of a missing command. Should we try to continue and finish the rest of the command sequence although one command won't happen, or should we abort as soon as possible and let the user know.

Reply:
It is probably better to abort. Most of these commands drive the ATV interface and if one step fails then the end result will almost always be wrong, so not worth progressing with the next commands e.g. if `home`, `left`, `down`, `select` is typed in as `home`, `left`, `done`, `select`, then the wrong item in the UI will be selected.

Reply:
Updated to log an error and to break out the loops

Reply:
I'm happy to add raising an error. In my use case it adds no value as I hide most of my automation behind Siri voice commands and just check the home assistant logs from time to time to see if anything untoward is happening :-)

Reply:
Those notifications don't stay on the screen for long. Do they.
I'll shortened the `ValueError` message to give the user a chance to read it :-)
Also, since the `ValueError` generates 56 lines of logs, is it worth logging an error too, or just go with `ValueError` ?

Reply:
We usually don't double up with log and raising an error in these cases. I suggest we just raise the error. The goal is to make it clear for the user what to correct and then the stack trace won't recur.